### ISSUE TITLE
Unhandled Error on creation of new project with example files

### ISSUE THREAD

**Issue by QuantumCowboy** [vkbo/novelWriter#933]
Hello, I recently installed novelWriter on Windows 10 in order to explore it a bit more. I like what I'm seeing so far! I did just encounter an error though. On creating a new project and selecting the option to populate with example files, I receive the following error.

I'm wondering if it is a problem with my own Python installation... 

Will continue exploring. Thanks! 
 

> Environment:
> novelWriter Version: 1.5.1
> Host OS: win32 (10.0.19043)
> Python: 3.9.9 (0x30909f0)
> Qt: 5.15.2, PyQt: 5.15.6
> 
> Error:
> Dict unusable: the underlying C-library object has been freed.
> 
> Traceback:
>   File "C:\Users\[..]\Documents\local\novelWriter-1.5.1-minimal-win\novelwriter\gui\dochighlight.py", line 392, in highlightBlock
>     if not self.spEnchant.checkWord(rxMatch.captured(0)):
> 
>   File "C:\Users\[..]\Documents\local\novelWriter-1.5.1-minimal-win\novelwriter\core\spellcheck.py", line 78, in checkWord
>     return self.theDict.check(theWord)
> 
>   File "C:\Users\[..]\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\enchant\__init__.py", line 622, in check
>     self._check_this()
> 
>   File "C:\Users\[..]\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\enchant\__init__.py", line 594, in _check_this
>     super()._check_this(msg)
> 
>   File "C:\Users\[..]\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\enchant\__init__.py", line 152, in _check_this
>     raise Error(msg)
> 
>

**Comment by QuantumCowboy** [comment]
Addendum: On closing the error dialog box, novelWriter simply closes. Attempting to reopen the same sample project results in the same error and close.

**Comment by vkbo** [comment]
Hmm, it seems to be crashing when looking up a word in the spell check dictionary. It doesn't seem to have anything to do with the project itself. Looking at the enchant source code, it seems to be an internal safeguard in case it cannot find the unerlying c library, although I'm not entirely sure:

https://github.com/pyenchant/pyenchant/blob/main/enchant/__init__.py#L149

Could you try running the following from a command line window to reinstall it? Just to check that it isn't your installation that is broken.
```
python -m pip install --force-reinstall enchant
```

I will add some error handling around the enchant library to prevent it from crashing novelWriter at least. If this is indeed an issue with the package, there isn't much I can fix other than add a way for novelWriter to handle it safely. This is one of the few performance critical parts of the code, but in Python 3.10 adding a try/except block around the call has no overhead cost any more, so it isn't a big problem.

If you cannot fix it, you can uninstall enchant, which will effectively disable spell checking.

**Comment by QuantumCowboy** [comment]
> PS C:\Users\[...]> python -m pip install --force-reinstall enchant
> ERROR: Could not find a version that satisfies the requirement enchant (from versions: none)
> ERROR: No matching distribution found for enchant

There might be something wrong with my python paths... I don't ever use python on this windows machine, but it is one I take around for travel / writing / casual use. 

I can start a few project okay. However, I tried turning on spellcheck in the menu of a new test project to see if that caused a problem... it does immediately cause the crash. And unfortunately, I can never open that project again, it just crashes on load, I assume because the spell check toggle is saved as on.

**Comment by QuantumCowboy** [comment]
And fishing around in the .nwx file for the spell check boolean, setting it to false allows me to open the project again. I'll just be very careful not to turn that on for now! :)

**Comment by vkbo** [comment]
Well, I will make a fix that ensures that enchant doesn't crash novelWriter in the future. You can safely uninstall the enchant package as novelWriter works fine without it.

The enchant package didn't use to work on Windows 64-bit at all, until someone took over the pyenchant project and updated it. I used to have an alternative spell check implementation that worked on Windows, but was a lot more basic. Perhaps I should add that implementation back in. The problem with it is that you have to provide the dictionaries yourself.

**Comment by vkbo** [comment]
> PS C:\Users[...]> python -m pip install --force-reinstall enchant
> ERROR: Could not find a version that satisfies the requirement enchant (from versions: none)
> ERROR: No matching distribution found for enchant

Oh, by the way, this was my mistake. The package is called pyenchant on pip. It should be:
```
python -m pip install --force-reinstall enchant
```
I just tested it on a Windows virtualbox. See if that works.

**Comment by vkbo** [comment]
> PS C:\Users[...]> python -m pip install --force-reinstall enchant
> ERROR: Could not find a version that satisfies the requirement enchant (from versions: none)
> ERROR: No matching distribution found for enchant

Oh, by the way, this was my mistake. The package is called pyenchant on pip. It should be:
```
python -m pip install --force-reinstall pyenchant
```
I just tested it on a Windows virtualbox. See if that works.

**Comment by vkbo** [comment]
I made a new release that adds safeguards around the calls to the enchant package that didn't already have them. I also had another minor fix that was useful to backport from 1.6.

**Comment by QuantumCowboy** [comment]
Thank you for the rapid response on this! I did try the force reinstall pyenchant, which looked successful, but still had the same issue on 1.5.1. On installing 1.5.2, the crashing is repaired! Toggling spellcheck on and re-running spellcheck both appear to do nothing, even with an intentionally gibberish word in the current file. I'm sure there is something wrong on my windows python paths. However, the program is running and that's fine with me! 

I'm going to try this program out for my new novel project. I do most of my drafting on a very old linux box, so everything is working fine there. I'll probably dump it to OpenOffice or something later for editing and revision, whenever I get to that stage, so spellcheck is really not needed for me right now. I'll mess around with my python install on the windows machine and see if I can fix whatever is messed up over there in the next few months. 

Thanks for your help and for your quick response! I'm enjoying your program.
